File: tests/fixtures/sample-multilang-app/Sample.java
// Sample Java file for Zaria multi-language fixture.
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.HashMap;

/**
 * Represents a record of key-value pairs.
 */
public class Record {
    private int id;
    private String name;
    private double value;

    public Record(int id, String name, double value) {
        this.id = id;
        this.name = name;
        this.value = value;
    }

    public int getId() { return id; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public double getValue() { return value; }
}

/**
 * Processes a list of records.
 */
public class DataProcessor {
    private final List<Record> records;

    public DataProcessor(List<Record> records) {
        this.records = new ArrayList<>(records);
    }

    public List<Record> process() {
        List<Record> result = new ArrayList<>();
        for (Record r : records) {
            r.setName(r.getName().trim());
            result.add(r);
        }
        return result;
    }

    public static String loadFile(Path path) throws IOException {
        return Files.readString(path);
    }

    public static void writeFile(Path path, String content) throws IOException {
        Files.writeString(path, content);
    }

    public static void main(String[] args) throws IOException {
        if (args.length < 1) {
            System.err.println("Usage: DataProcessor <path>");
            System.exit(1);
        }
        String content = loadFile(Path.of(args[0]));
        System.out.println(content.substring(0, Math.min(80, content.length())));
    }
}

File: tests/fixtures/sample-multilang-app/Sample.cs
// Sample C# file for Zaria multi-language fixture.
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace ZariaFixture
{
    public record Record(int Id, string Name, double Value);

    public interface IProcessor
    {
        IEnumerable<Record> Process(IEnumerable<Record> records);
    }

    public class TrimProcessor : IProcessor
    {
        public IEnumerable<Record> Process(IEnumerable<Record> records)
        {
            return records.Select(r => r with { Name = r.Name.Trim() });
        }
    }

    public static class FileHelper
    {
        public static string LoadFile(string path) => File.ReadAllText(path);

        public static void WriteFile(string path, string content) =>
            File.WriteAllText(path, content);

        public static async Task<string> LoadFileAsync(string path) =>
            await File.ReadAllTextAsync(path);
    }

    public class DataProcessor
    {
        private readonly List<Record> _records;

        public DataProcessor(IEnumerable<Record> records)
        {
            _records = new List<Record>(records);
        }

        public List<Record> Process()
        {
            return _records
                .Select(r => r with { Name = r.Name.Trim() })
                .ToList();
        }

        public static async Task Main(string[] args)
        {
            if (args.Length < 1)
            {
                Console.Error.WriteLine("Usage: sample <path>");
                Environment.Exit(1);
            }
            var content = await FileHelper.LoadFileAsync(args[0]);
            Console.WriteLine(content[..Math.Min(80, content.Length)]);
        }
    }
}
